src/instructlab/configuration.py

```diff
@@ -1207,6 +1208,57 @@ def recreate_train_profiles(overwrite: bool = False) -> bool:
     return fresh_install
 
 
+# read_and_create_system_profiles walks the given dir, reads the templated files and writes them into
+# the DEFAULTS.SYSTEM_PROFILE_DIR preserving their arch, processor name, and yaml file name
+# pylint: disable=broad-exception-caught
+def read_and_create_system_profiles(profiles_dir: str, overwrite: bool) -> bool:
+    fresh_install = False
```

Comment:
Please add a docstring to this func so folks doing future work will have some context

Reply:
The above test could probably be used for this